datafusion/physical-plan/src/sorts/sort.rs

```diff
@@ -645,7 +645,36 @@ impl ExternalSorter {
             return self.sort_batch_stream(batch, metrics, reservation);
         }
 
-        let streams = std::mem::take(&mut self.in_mem_batches)
+        let mut merged_batches = Vec::new();
+        let mut current_batches = Vec::new();
+        let mut current_size = 0;
+
+        for batch in std::mem::take(&mut self.in_mem_batches) {
```

Comment:
I think it would be nice to use `pop` (`while let Some(batch) = v.pop`) here to remove the batch from the vec once sorted to reduce memory usage. Now the batch is AFAIK retained until after the loop.

Reply:
> I think it would be nice to use `pop` (`while let Some(batch) = v.pop`) here to remove the batch from the vec once sorted to reduce memory usage. Now the batch is AFAIK retained until after the loop.

Thank you @Dandandan for review and good suggestion, addressed your suggestion!